source/NationalInstruments/jquery.flot.highlights.js

```diff
@@ -0,0 +1,210 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+/* Flot plugin for adding axis handle to the graph
+*/
+
+/*global jQuery*/
+
+(function ($) {
+    'use strict';
+
+    class Highlights {
+        static get defaultOptions() {
+            return {
+                series: {
+                    highlights: {
+                        selectedRange: [],
+                        selectedIndexes: [],
+                        show: false,
+                        lineWidth: 1,
+                        highlightColor: '#ffffff',
+                        highlightPoints: true,
+                        highlightLines: false,
+                        highlightBars: false,
+                    }
+                }
+            }
+        }
+
+        constructor(plot, options) {
+            this._processOptions(options);
+            this._createHooks(plot);
+        }
+
+        _createHooks(plot) {
+            plot.hooks.drawOverlay.push((plot, ctx, overlay) => {
+                this._drawOverlay(plot, ctx, overlay)
+            });
+        }
+
+        _processOptions(options) {
+            $.extend(true, options, this.defaultOptions);
+            this._options = options.series.highlights;
+        }
+
+        _drawOverlay(plot, ctx) {
+            const series = plot.getData();
+            if (!this._options.show) {
+                return;
+            }
+
+            const plotOffset = plot.getPlotOffset();
+            ctx.save();
+            ctx.translate(plotOffset.left, plotOffset.top);
+            ctx.lineWidth = this._options.lineWidth;
+            ctx.lineCap = 'round';
+            ctx.lineJoin = 'round';
+            let lastPointIndex;
+            for (let j = 0; j < series.length; j++) {
+                let ps = series[j].datapoints.pointsize;
+                let highlightIndexes = this._options.selectedIndexes[j];
+                if (highlightIndexes) {
+                    for (let i = 0; i < highlightIndexes.length; i++) {
+                        let pointIndex = highlightIndexes[i];
+                        if (this._options.highlightPoints) {
+                            this._drawPointHighlight(series[j], plot, ctx, pointIndex);
+                        }
+
+                        if (this._options.highlightBars) {
+                            this._drawBarHighlight(series[j], [series[j].datapoints.points[pointIndex * ps], series[j].datapoints.points[pointIndex * ps + 1]], ctx);
+                        }
+                    }
+                }
+
+                let highlightRange = this._options.selectedRange[j];
+                if (highlightRange) {
```

Comment:
I guess I had originally expected an either/or handling of highlighting indexes vs. range, but I think you're right that handling both shouldn't be disallowed (and could be useful).